### ISSUE TITLE
Unable to create a variable of type 'VectorOfInt' required for call to imgcodecs::imwrite

### ISSUE THREAD

**Issue by svenne-juul** [twistedfall/opencv-rust#361]
The call of imgcodecs::imwrite (for saving of an image) requires passing of a 'VectorOfint' containing various parameters affecting how the image is saved. VecorOfint is according to the documentation in either 'opencv::prelude' or before that in 'opencv::types' but compilation fails when attempting to create a variable of the type. 

**error[E0433]: failed to resolve: could not find `VectorOfint` in `prelude`
  --> src\main.rs:15:44
   |
15 |     let mut paramsarray = opencv::prelude::VectorOfint::new();
   |                                            ^^^^^^^^^^^ could not find `VectorOfint` in `prelude`**

```
use opencv::{
    prelude::*,
    imgcodecs,
    types,
};
fn run() -> opencv::Result<()> {
    let filename = "Philly_1mb.jpg";
    let window_a: &str = filename.clone();
    let mut mat = imgcodecs::imread(&filename, imgcodecs::IMREAD_COLOR)?;
    if mat.size()?.width == 0 {
        panic!("Unable to read image a file");
        };
    println!("Image a data: {:?}",mat);
// Failing statement below
_**//    let mut paramsarray = opencv::types::VectorOfint::new();
    let mut paramsarray = opencv::prelude::VectorOfint::new();**_
// Failing statement above
    let outputfilename = "Philly_1mb_b.jpg";
    imgcodecs::imwrite(&outputfilename,&mat,&paramsarray)?;
    Ok(())
    }   


fn main() {
    run().unwrap()
}

```
Operating system:
Edition Windows 10 Pro
Version 21H2
Installed on ‎8/‎10/‎2022
OS build 19044.1889
Experience Windows Feature Experience Pack 120.2212.4180.0

Installation
Installed using the approach relying on [chocolatey](https://chocolatey.org/) described under 'Windows package'. I believe the install works correctly as many other opencv-related sample programs work just fine.

3. OpenCV version
4. rustc version (`rustc --version`)
5. Attach the full output of the following command from your project directory:
   ```shell script
   RUST_BACKTRACE=full cargo build -vv 
PS C:\Users\svenn\rust\pxlit_opencv> cargo build -vv
       Fresh autocfg v1.1.0
       Fresh cfg-if v1.0.0
       Fresh once_cell v1.13.0
       Fresh scopeguard v1.1.0
       Fresh glob v0.3.0
       Fresh unicode-ident v1.0.3
       Fresh num_cpus v1.13.1
       Fresh adler32 v1.2.0
       Fresh regex-syntax v0.6.27
       Fresh adler v1.0.2
       Fresh either v1.8.0
       Fresh maplit v1.0.2
       Fresh dunce v1.0.2
       Fresh percent-encoding v2.1.0
       Fresh jobserver v0.1.24
       Fresh weezl v0.1.7
       Fresh shlex v1.1.0
       Fresh vcpkg v0.2.15
       Fresh futures-sink v0.3.23
       Fresh pkg-config v0.3.25
       Fresh color_quant v1.1.0
       Fresh bitflags v1.3.2
       Fresh lebe v0.5.2
       Fresh rawpointer v0.2.1
       Fresh smallvec v1.9.0
       Fresh half v1.8.2
       Fresh bit_field v0.10.1
       Fresh scoped_threadpool v0.1.9
       Fresh byteorder v1.4.3
       Fresh bytemuck v1.12.1
       Fresh getrandom v0.2.7
       Fresh threadpool v1.8.1
       Fresh miniz_oxide v0.5.3
       Fresh deflate v1.0.0
       Fresh inflate v0.4.5
       Fresh cc v1.0.73
       Fresh gif v0.11.4
       Fresh matrixmultiply v0.3.2
       Fresh crossbeam-utils v0.8.11
       Fresh winapi v0.3.9
       Fresh proc-macro2 v1.0.43
       Fresh libc v0.2.129
       Fresh nanorand v0.7.0
       Fresh memchr v2.5.0
       Fresh crc32fast v1.3.2
       Fresh semver v1.0.13
       Fresh futures-core v0.3.23
       Fresh memoffset v0.6.5
       Fresh num-traits v0.2.15
       Fresh crossbeam-channel v0.5.6
       Fresh lock_api v0.4.7
       Fresh libloading v0.7.3
       Fresh quote v1.0.21
       Fresh aho-corasick v0.7.18
       Fresh flate2 v1.0.24
       Fresh png v0.17.5
       Fresh crossbeam-epoch v0.9.10
       Fresh spin v0.9.4
       Fresh num-integer v0.1.45
       Fresh num-complex v0.4.2
       Fresh clang-sys v1.3.3
       Fresh syn v1.0.99
       Fresh regex v1.6.0
       Fresh crossbeam-deque v0.8.2
       Fresh num-rational v0.4.1
       Fresh clang v2.0.0
       Fresh pin-project-internal v1.0.12
       Fresh ndarray v0.15.6
       Fresh rayon-core v1.9.3
       Fresh pin-project v1.0.12
       Fresh opencv-binding-generator v0.45.0
       Fresh rayon v1.5.3
       Fresh flume v0.10.14
       Fresh jpeg-decoder v0.2.6
       Fresh exr v1.4.2
       Fresh tiff v0.7.3
       Fresh opencv v0.66.0
       Fresh image v0.24.3
   Compiling pxlit_opencv v0.1.0 (C:\Users\svenn\rust\pxlit_opencv)
     Running `set CARGO=\\?\C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe&& set CARGO_BIN_NAME=pxlit_opencv&& set CARGO_CRATE_NAME=pxlit_opencv&& set CARGO_MANIFEST_DIR=C:\Users\svenn\rust\pxlit_opencv&& set CARGO_PKG_AUTHORS=""&& set CARGO_PKG_DESCRIPTION=""&& set CARGO_PKG_HOMEPAGE=""&& set CARGO_PKG_LICENSE=""&& set CARGO_PKG_LICENSE_FILE=""&& set CARGO_PKG_NAME=pxlit_opencv&& set CARGO_PKG_REPOSITORY=""&& set CARGO_PKG_VERSION=0.1.0&& set CARGO_PKG_VERSION_MAJOR=0&& set CARGO_PKG_VERSION_MINOR=1&& set CARGO_PKG_VERSION_PATCH=0&& set CARGO_PKG_VERSION_PRE=""&& set CARGO_PRIMARY_PACKAGE=1&& set PATH="C:\Users\svenn\rust\pxlit_opencv\target\debug\deps;C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;C:\Program Files\Dell\DW WLAN Card;;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\WIDCOMM\Bluetooth Software\;C:\Program Files\WIDCOMM\Bluetooth Software\syswow64;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft VS Code\bin;C:\ProgramData\chocolatey\bin;C:\Program Files\LLVM\bin;C:\Users\svenn\.cargo\bin;C:\Users\svenn\AppData\Local\Microsoft\WindowsApps;"&& rustc --crate-name pxlit_opencv --edition=2021 src\main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=48943ec9e4d1a38f --out-dir C:\Users\svenn\rust\pxlit_opencv\target\debug\deps -C incremental=C:\Users\svenn\rust\pxlit_opencv\target\debug\incremental -L dependency=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps --extern image=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libimage-bd9c5d2de8a0ecf7.rlib --extern ndarray=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libndarray-6af201cef50c369f.rlib --extern opencv=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libopencv-984bef83365d79f6.rlib -L native=C:\Users\svenn\rust\pxlit_opencv\target\debug\build\opencv-e0385181a2a64e23\out -L C:\tools\opencv\build\x64\vc15\lib`
error[E0433]: failed to resolve: could not find `VectorOfint` in `prelude`
  --> src\main.rs:15:44
   |
15 |     let mut paramsarray = opencv::prelude::VectorOfint::new();
   |                                            ^^^^^^^^^^^ could not find `VectorOfint` in `prelude`

warning: unused import: `types`
 --> src\main.rs:4:5
  |
4 |     types,
  |     ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0433`.
warning: `pxlit_opencv` (bin "pxlit_opencv") generated 1 warning
error: could not compile `pxlit_opencv` due to previous error; 1 warning emitted

Caused by:
  process didn't exit successfully: `set CARGO=\\?\C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin\cargo.exe&& set CARGO_BIN_NAME=pxlit_opencv&& set CARGO_CRATE_NAME=pxlit_opencv&& set CARGO_MANIFEST_DIR=C:\Users\svenn\rust\pxlit_opencv&& set CARGO_PKG_AUTHORS=""&& set CARGO_PKG_DESCRIPTION=""&& set CARGO_PKG_HOMEPAGE=""&& set CARGO_PKG_LICENSE=""&& set CARGO_PKG_LICENSE_FILE=""&& set CARGO_PKG_NAME=pxlit_opencv&& set CARGO_PKG_REPOSITORY=""&& set CARGO_PKG_VERSION=0.1.0&& set CARGO_PKG_VERSION_MAJOR=0&& set CARGO_PKG_VERSION_MINOR=1&& set CARGO_PKG_VERSION_PATCH=0&& set CARGO_PKG_VERSION_PRE=""&& set CARGO_PRIMARY_PACKAGE=1&& set PATH="C:\Users\svenn\rust\pxlit_opencv\target\debug\deps;C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;C:\Users\svenn\.rustup\toolchains\stable-x86_64-pc-windows-msvc\bin;C:\Program Files\Dell\DW WLAN Card;;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\WIDCOMM\Bluetooth Software\;C:\Program Files\WIDCOMM\Bluetooth Software\syswow64;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Microsoft VS Code\bin;C:\ProgramData\chocolatey\bin;C:\Program Files\LLVM\bin;C:\Users\svenn\.cargo\bin;C:\Users\svenn\AppData\Local\Microsoft\WindowsApps;"&& rustc --crate-name pxlit_opencv --edition=2021 src\main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 -C metadata=48943ec9e4d1a38f --out-dir C:\Users\svenn\rust\pxlit_opencv\target\debug\deps -C incremental=C:\Users\svenn\rust\pxlit_opencv\target\debug\incremental -L dependency=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps --extern image=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libimage-bd9c5d2de8a0ecf7.rlib --extern ndarray=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libndarray-6af201cef50c369f.rlib --extern opencv=C:\Users\svenn\rust\pxlit_opencv\target\debug\deps\libopencv-984bef83365d79f6.rlib -L native=C:\Users\svenn\rust\pxlit_opencv\target\debug\build\opencv-e0385181a2a64e23\out -L C:\tools\opencv\build\x64\vc15\lib` (exit code: 1)
PS C:\Users\svenn\rust\pxlit_opencv>   ```

Here is a test image that can be used once the compilation issue is resolved.
![Philly_1mb](https://user-images.githubusercontent.com/111534511/185920087-8f7a23c1-f64b-4c48-94c0-f4a74ed43c40.jpg)

**Comment by twistedfall** [comment]
You should use `VectorOfi32` (https://docs.rs/opencv/latest/opencv/types/type.VectorOfi32.html). Can you please tell me where does the name `VectorOfint` coming from? It’s an old alias and it was removed quite a while ago so I might need to update some docs.

**Comment by svenne-juul** [comment]
Thank you again for your help. I googled 'rust imgcodecs::imwrite'  and that took me to page a [doc page](https://docs.rs/opencv/0.8.0/opencv/imgcodecs/fn.imwrite.html) that I just now realize was outdated. Totally my error!. I guess google gave the outdated page higher ranking in the search results than the newer page because it had been around longer. I don't think I will make THAT mistake again.
When I go to the new page I do indeed see the 'i32' 
With this correction (paramsarray = opencv::core::Vector::<i32>::new();) the rest of the logic ran fine and I got my output file.
As far as I am concerned I can close this issue unless you see that differently.
